src/irmin/tree_intf.ml

```diff
@@ -274,22 +275,30 @@ module type S = sig
     ?force:'a force ->
     ?cache:bool ->
     ?uniq:uniq ->
-    ?pre:'a node_fn ->
-    ?post:'a node_fn ->
+    ?pre:('a, step list) folder ->
+    ?post:('a, step list) folder ->
     ?depth:depth ->
-    ?contents:(path -> contents -> 'a -> 'a Lwt.t) ->
-    ?node:(path -> node -> 'a -> 'a Lwt.t) ->
-    ?tree:(path -> t -> 'a -> 'a Lwt.t) ->
+    ?contents:('a, contents) folder ->
+    ?node:('a, node) folder ->
+    ?tree:('a, t) folder ->
     t ->
     'a ->
     'a Lwt.t
-  (** [fold f t acc] folds [f] over [t]'s leafs.
+  (** [fold t acc] folds over [t]'s nodes with node-specific folders:
+      [contents], [node], and [tree], based on a node's {!kind}.
 
-      For every node [n], ui [n] is a leaf node, call [f path n]. Otherwise:
+      The default for all folders is identity.
 
-      - Call [pre path n]. By default [pre] is the identity;
-      - Recursively call [fold] on each children.
-      - Call [post path n]; By default [post] is the identity.
+      For every node [n] of [t], including itself:
+
+      - If [n] is a [`Contents] kind, call [contents path c] where [c] is the
+        {!contents} of [n].
+      - If [n] is a [`Node] kind, (1) call [pre path steps]; (2) call
+        [node path n]; (3) recursively fold on each child; (4) call
+        [post path steps].
+      - If [n] is any kind, call [tree path t'] where [t'] is the tree of [n].
+
+      See [examples/fold.ml] for a demo of the different {!folder}s.
```

Comment:
It would be nice to provide a link the the fold example for people reading the generated documentation outside of the Github repo:

```suggestion
      See {{: https://github.com/mirage/irmin/blob/main/examples/deploy.ml} examples/fold.ml} for a demo of the different {!folder}s.
```

Reply:
Good idea. I'll make the change locally to ensure proper formatting.